### ISSUE TITLE
Using orderBy in firestoreConnect query prevents results from being returned

### ISSUE THREAD

**Issue by GitHubGreg** [prescottprue/redux-firestore#215]
**What is the current behavior?**
When I run a query using firestoreConnect with a `where` and an `orderBy`, it fails. If I remove the `orderBy` it passes. The console doesn't offer me the link to create the index, so I tried manually creating it but it still fails to return the firestore collection. My issue is identical to [this](https://github.com/prescottprue/redux-firestore/issues/50).

**What is the expected behavior?**
Return the firestore collection specified by the `where` and an `orderBy` query.

**Which version of redux-firestore are you using? What about other dependencies?**
```
{
    "firebase": "^5.3.0",
    "react-dom": "^16.7.0",
    "react-redux": "5.0.7",
    "react-redux-firebase": "^2.2.6",
    "redux": "4.0.0",
    "redux-firestore": "^0.7.0",
    "redux-saga": "^0.16.0",
  }
```

**Which environments/browsers are affected by this issue? Did this work in previous versions or setups?**
I have tried in Firefox, Chrome and Safari on Mac. I have never been able to get this to work in three projects using redux-firestore.

**Minimal demo to reproduce issue (using [codesandbox](https://codesandbox.io/) or similar)**
[Here is my simplified code gist](https://gist.github.com/GitHubGreg/e3d7aee012c5eca49bd3fbd66e344bfc)

**Comment by illuminist** [comment]
Works perfectly fine for me. Just make sure you have been creating an index like this
![image](https://user-images.githubusercontent.com/1757315/57833140-93a73080-77e3-11e9-8be3-b25fb8db0c49.png)
![image](https://user-images.githubusercontent.com/1757315/57833378-13cd9600-77e4-11e9-9183-d6c2ac8606a3.png)

My reproduction https://codesandbox.io/s/reactreduxfirebase-reduxfirestoreissues215-9od6d
You can query for year 2000 and 2001.

**Comment by GitHubGreg** [comment]
Thanks for looking into this so rigorously. On my end, I already have that exact index (see below) but no results returned. That said, the console message with URL to create the index never appeared (so I created it manually).

Good question. Yes, when I use the firebase package directly it works and returns the results. 

Here is my code for what it's worth. Like yours but more verbose.

```
      let ref = await firestore.collection('membershipDefs');
      console.log('TCL: MembershipDefs -> ref', ref);
      let query = await ref.where('year', '==', '2019');
      console.log('TCL: MembershipDefs -> query', query);
      let querySnap = await query.get();
      console.log('TCL: MembershipDefs -> querySnap', querySnap);
      for (let i = 0; i < querySnap.docs.length; i++) {
        console.log('index ' + i + ': ');
        console.log(querySnap.docs[i].data());
      }

```

**Comment by illuminist** [comment]
What is datatype of `year`? The query datatype need to match both of query and database. I modified my repro to have year as a string and it's still working.

Your latest code is missing `.orderBy('key', 'asc')` which might be an important clue why your code doesn't work.

Could you fork my sandbox and modify until it doesn't work?

**Comment by GitHubGreg** [comment]
Good point. I added `.orderBy('key', 'asc')` to my above firestore query and it gave me an error to create the index. I clicked the link and the only difference between the one I had manually made previously and the one it wanted to make was the order. See below.

![Screen Shot 2019-05-20 at 7 27 08 PM](https://user-images.githubusercontent.com/7059106/58058134-aef0a380-7b35-11e9-94ae-28146ad3ae3f.jpg)

It therefore seems that, even if one condition is a "where" and the other is an "orderBy", the order matters when making the composite index.

So problem solved. Thanks again for your persistence in helping resolve this.
